pop-api/integration-tests/contracts/fungibles/lib.rs

```diff
@@ -61,7 +64,13 @@ mod fungibles {
 
 		#[ink(message)]
 		pub fn transfer(&mut self, id: AssetId, to: AccountId, value: Balance) -> Result<()> {
-			api::transfer(id, to, value)
+			let result = api::transfer(id, to, value);
+			self.env().emit_event(Transfer {
```

Comment:
Why emit the event if the transfer failed? Should be transfer(...)?; and then Ok(()) surely?

Reply:
Same with all the others.